js_modules/dagster-ui/packages/ui-core/src/assets/AssetTabs.tsx

```diff
@@ -65,7 +65,7 @@ export const buildAssetTabMap = (input: AssetTabConfigInput) => {
       id: 'partitions',
       title: 'Partitions',
       to: buildAssetViewParams({...params, view: 'partitions'}),
-      hidden: !definition?.partitionDefinition || definition?.isSource,
+      hidden: !definition?.partitionDefinition || !definition?.isMaterializable,
```

Comment:
Technically this also hides it if the definition does not exist, I think in this case we may want to say `(definition && !definition.isMaterializable)` so that the `!definition` case doesn't also count

Reply:
Gah I accidentally merged before fixing this. Here's a followup: https://github.com/dagster-io/dagster/pull/23490.

Reply:
Actually – thinking about this more deeply after noticing a test failure on my followup PR branch, I think the expected behavior is actually that we _do_ want to hide the tab if the definition does not exist. So I think this is right?